node/src/services/file_transfer/handler.rs

```diff
@@ -146,12 +146,52 @@ impl Actor for FileTransferService {
                 FileTransferServiceCommand::AddKnownAddress {
                     peer_id,
                     multiaddress,
-                } => self.network.add_known_address(peer_id, multiaddress),
-                FileTransferServiceCommand::RegisterNewFile { peer_id, file_key } => {
-                    self.peer_file_registry.insert((peer_id, file_key));
+                    callback,
+                } => {
+                    self.network.add_known_address(peer_id, multiaddress);
+                    // `add_known_address()` method doesn't return anything.
+                    match callback.send(Ok(())) {
+                        Ok(()) => {}
+                        Err(_) => error!(
+                            target: LOG_TARGET,
+                            "Failed to send the response back. Looks like the requester task is gone."
+                        ),
+                    }
+                }
+                FileTransferServiceCommand::RegisterNewFile {
+                    peer_id,
+                    file_key,
+                    callback,
+                } => {
+                    let result = match self.peer_file_registry.insert((peer_id, file_key)) {
+                        true => Ok(()),
+                        false => Err(RequestError::RegisterNewFileFailure),
```

Comment:
I think it's better to be specific with what the error means. Do you think of `PeerFileAlreadyRegistered`?